golem-rib/src/parser/rib_expr.rs

```diff
@@ -12,180 +12,1187 @@
+                    Expr::identifier_global("bar", None),
+                    Expr::identifier_global("baz", None),
+                ],
+            ),
+        ]);
+        assert_eq!(expr, expected);
+    }
+
+    #[test]
+    fn test_worker_function_invoke_6() {
+        let rib_expr = r#"
+          let worker = instance("my-worker");
+          worker.function-name[foo](foo, bar, baz)
+        "#;
+        let expr = Expr::from_text(rib_expr).unwrap();
+        let worker_variable = Expr::identifier_global("worker", None);
+        let function_name = "function-name".to_string();
+        let type_parameter = GenericTypeParameter {
+            value: "foo".to_string(),
+        };
+
+        let expected = Expr::expr_block(vec![
+            Expr::let_binding(
+                "worker",
+                Expr::call_worker_function(
+                    DynamicParsedFunctionName::parse("instance").unwrap(),
+                    None,
+                    None,
+                    vec![Expr::literal("my-worker")],
+                ),
+                None,
+            ),
+            Expr::invoke_worker_function(
+                worker_variable,
+                function_name,
+                Some(type_parameter),
+                vec![
+                    Expr::identifier_global("foo", None),
+                    Expr::identifier_global("bar", None),
+                    Expr::identifier_global("baz", None),
+                ],
+            ),
+        ]);
+        assert_eq!(expr, expected);
+    }
+
+    #[test]
+    fn test_worker_function_invoke_7() {
+        let rib_expr = r#"
+          let worker = instance[foo]("my-worker");
+          worker.function-name[bar](foo, bar, baz)
+        "#;
+        let expr = Expr::from_text(rib_expr).unwrap();
+        let worker_variable = Expr::identifier_global("worker", None);
+        let function_name = "function-name".to_string();
+        let type_parameter1 = GenericTypeParameter {
+            value: "foo".to_string(),
+        };
+
+        let type_parameter2 = GenericTypeParameter {
+            value: "bar".to_string(),
+        };
+
+        let expected = Expr::expr_block(vec![
+            Expr::let_binding(
+                "worker",
+                Expr::call_worker_function(
+                    DynamicParsedFunctionName::parse("instance").unwrap(),
+                    Some(type_parameter1),
+                    None,
+                    vec![Expr::literal("my-worker")],
+                ),
+                None,
+            ),
+            Expr::invoke_worker_function(
+                worker_variable,
+                function_name,
+                Some(type_parameter2),
+                vec![
+                    Expr::identifier_global("foo", None),
+                    Expr::identifier_global("bar", None),
+                    Expr::identifier_global("baz", None),
+                ],
+            ),
+        ]);
+        assert_eq!(expr, expected);
```

Comment:
90% of these tests are existing tests, as select_field.rs and select_index.rs files are deleted. No existing tests are deleted.